lib/matplotlib/stackplot.py

```diff
@@ -68,7 +68,8 @@ def stackplot(axes, x, *args, **kwargs):
 
     baseline = kwargs.pop('baseline', 'zero')
     # Assume data passed has not been 'stacked', so stack it here.
-    stack = np.cumsum(y, axis=0)
+    # We'll need a float buffer for the upcoming calculations.
+    stack = np.cumsum(y, axis=0).astype(np.promote_types(y.dtype, np.float32))
```

Comment:
I have no clue what it does under the hood, so it might be a performance improvement or just aesthetic improvement. It might avoid a memory allocation, but I wouldn't bet on it.